libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+        public itemEl: ElementRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm,
+        @Optional() @SkipSelf() @Host() formField: FormField,
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>,
+        protected _listConfig?: ListConfig) {
+        super(_changeDetectorRef, ngControl, ngForm, formField, formControl);
+    }
+
+    /** @hidden */
+    /** Instailization of list with selection mode*/
+    ngOnInit(): void {
+
+        if (this.dsItems.length !== null && this.itemSize !== 0) {
+            this.startIndex = 0;
+            this.lastIndex = this.itemSize;
+            this.items = [];
+            this.items = this.dsItems.slice(this.startIndex, this.lastIndex);
+        } else {
+            this.items = this.dsItems;
+        }
+        this.id = `fdp-list-${nextListId++}`;
+        // using selection Model for multiselect
+        if (this.selectionMode === 'multi') {
+            this.multiSelect = true;
+            this.ariaMultiselectable = true;
+        } else { this.multiSelect = false; }
+        this.selectionModel = new SelectionModel<BaseListItem>(
+            this.multiSelect,
+            this.selectedItems
+        );
+
+        this.selectionModel.changed.subscribe(e => {
+            this.selectedItems = this.selectionModel.selected;
+            const event = new SelectionChangeEvent();
+            event.selectedItems = this.selectedItems;
+            this.selectedItemChange.emit(event);
+        });
+
+
+    }
+
+    /** @hidden */
+    /**Keyboard manager on list items, set values when passed via array */
+    ngAfterViewInit(): void {
+
+        this.keyManager = new FocusKeyManager<BaseListItem>(this.ListItems).withWrap();
+        this.ListItems.forEach((item) => {
+            item.navigated = this._navigated;
+            item.navigationIndicator = this._navigationIndicator;
+            item.contentDensity = this.contentDensity;
+            item.selectionMode = this.selectionMode;
+            item.listType = this.listType;
+            item.hasByLine = this._hasByLine;
+            item.noSeperator = this.noSeperator;
+        });
+    }
+
+
+    /** @hidden */
+    /**Setting values from list to list items
+     * example:
+     * Does list item has navigation,
+     * should show arrows,
+     * will it be compact mode,
+     * should be in which selection mode
+     * set values when passed via datasource
+     */
+    ngAfterContentInit(): void {
+        this._itemsSubscription = this.ListItems.changes.subscribe((items) => {
+            items.forEach((item) => {
+                item.navigated = this._navigated;
+                item.navigationIndicator = this._navigationIndicator;
+                item.contentDensity = this.contentDensity;
+                item.selectionMode = this.selectionMode;
+                item.listType = this.listType;
+                item.hasByLine = this._hasByLine;
```

Comment:
the same as above emit event to parent

Reply:
updated here and in all applicable places for list.componen.ts